src/main/java/clod/operations/TaskList.java

```diff
@@ -0,0 +1,199 @@
+        updateTaskStatus(taskIndex, false);
+    }
+
+    private void updateTaskStatus(int taskIndex, boolean isDone) {
+        if (!isValidTaskIndex(taskIndex)) {
+            Interactions.printMessage("Come on, even I know that's not a valid task number.");
+            return;
+        }
+        Task task = tasks.get(taskIndex - 1);
+        task.setDone(isDone);
+        saveTaskToStorage();
+
+        if (isDone) {
+            Interactions.printMessage("Congrats for finishing this... I think?" + "\n" + task.getDescription());
+        } else
+            Interactions.printMessage("I guess you didn't finish this after all. Well, that's life I guess." + "\n" + task.getDescription());
+    }
+
+    private boolean isValidTaskIndex(int taskIndex) {
+        return taskIndex >= LIST_LOWER_BOUND && taskIndex <= tasks.size();
+    }
+
+    /**
+     * Displays all tasks in the list to the user.
+     * If the list is empty, provides appropriate feedback.
+     */
+    public void printList() {
+        if (tasks.isEmpty()) {
+            Interactions.printMessage("Hmm... The list seems empty. Did you say anything yet?");
+            return;
+        }
+        System.out.println(LIST_SEPARATOR);
+        Interactions.printMessage("Here's what you've said so far:");
+        printTasksInOrder();
+        System.out.println(LIST_SEPARATOR);
+    }
+
+    private void printTasksInOrder() {
+        for (int i = 0; i < tasks.size(); i++) {
+            System.out.println((i + 1) + ". " + tasks.get(i).getDescription());
+        }
+    }
+
+    public void findTasksByKeyword(String keyword) {
+        if (tasks.isEmpty()) {
+            Interactions.printMessage("There's nothing to search through. The list is emptier than my... well, everything.");
+            return;
+        }
+
+        List<Task> matchingTasks = new ArrayList<>();
+        for (Task task : tasks) {
+            String taskDescription = task.getDescription().toLowerCase();
+            if (taskDescription.contains(keyword)) {
+                String description = task.getDescription().toLowerCase();
+                if (description.contains(keyword.toLowerCase())) {
+                    matchingTasks.add(task);
+                }
+            }
+
+            if (matchingTasks.isEmpty()) {
+                Interactions.printMessage("Searched high and low but couldn't find '" + keyword + "'. " +
+                        "Are you sure it exists? I'm not exactly known for my memory, you know.");
+                return;
+            }
+
+            // Print the matching tasks
+            System.out.println(LIST_SEPARATOR);
+            Interactions.printMessage("Here are the matching tasks in your list:");
+            for (int i = 0; i < matchingTasks.size(); i++) {
+                System.out.println((i + 1) + ". " + matchingTasks.get(i).getDescription());
+            }
+            System.out.println(LIST_SEPARATOR);
+        }
+    }
+}
```

Comment:
consider separating out task related methods to a different class rather than putting it under task list